crates/network-scanner/src/interfaces/mod.rs

```diff
@@ -0,0 +1,27 @@
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "windows")]
+mod windows;
+
+#[cfg(target_os = "windows")]
+pub use windows::get_network_interfaces;
+
+#[cfg(target_os = "linux")]
+pub use linux::get_network_interfaces;
+
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
serialize to be used later in gateway

Reply:
I would prefer we don’t reuse this struct as a "API model" (think [DTO](https://en.wikipedia.org/wiki/Data_transfer_object)), and more generally that we don’t define this kind of "system boundary API" too far away from the actual system boundary, in this case, the HTTP endpoint of the Devolutions Gateway.

Instead, I suggest you treat this struct as an internal/business model for the `network-scanner` crate, and then later define a struct specifically to serve as a "API model" directly in the `devolutions_gateway::api::network_scan` module.

This way, you don’t have to think about the system boundaries of the Devolutions Gateway when working on this crate. The [rules at the boundaries are different](https://www.tedinski.com/2018/02/06/system-boundaries.html). Let’s leave serialization concerns where they belong.

Reply:
understand, updated!